src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -113,10 +113,10 @@ private Task OnProjectFactoryCompletedAsync()
             // TODO: https://github.com/dotnet/roslyn-project-system/issues/353
             await _commonServices.ThreadingService.SwitchToUIThread();
 
-            using (_tasksService.LoadedProject())
+            await _tasksService.LoadedProjectAsync(async () =>
```

Comment:
Does LoadedProjectAsync ensure that passed in task is executed on the current thread context?